internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -417,6 +422,73 @@ func (cf CloudFormation) createEnvControllerRenderer(in *envControllerRendererIn
 	}), nil
 }
 
+type renderStackInput struct {
+	group *errgroup.Group // Group of go routines.
+	ctx   context.Context // Context shared between go routines.
+
+	// Stack metadata.
+	stackName      string            // Name of the stack.
+	stackID        string            // ID of the stack.
+	description    string            // Descriptive text for the stack mutation.
+	descriptionFor map[string]string // Descriptive text for each resource in the stack.
+	startTime      time.Time         // Timestamp for when the stack mutation started.
+}
+
+func (cf CloudFormation) stackRenderer(in renderStackInput) progress.DynamicRenderer {
```

Comment:
https://pkg.go.dev/context
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx
```suggestion
func (cf CloudFormation) stackRenderer(ctx context.Context, in renderStackInput) progress.DynamicRenderer {
```

Reply:
Done!